src/extension/sldService/src/main/java/org/geoserver/sldservice/utils/classifier/RulesBuilder.java

```diff
@@ -499,4 +531,45 @@ private Filter getNotOverlappingFilter(
 
         return f;
     }
+
+    private class PercentagesManager {
+
+        private double[] percentages;
+        List<Double> collapsedPercentages;
+        boolean collapsePercentages;
+
+        public PercentagesManager(double[] percentages) {
+            this.percentages = percentages;
+            this.collapsePercentages =
+                    percentages != null && !(DoubleStream.of(percentages).sum() > 100.0);
+            this.collapsedPercentages = new ArrayList<>();
+        }
+
+        public void collapsePercentages(List<Rule> rules, Rule current, int index) {
+            if (outputPercentages && collapsePercentages) {
+                int ruleIdx = rules.indexOf(current);
+                double toSumTo = percentages[index];
+                double toBeSummed = collapsedPercentages.get(ruleIdx);
+                collapsedPercentages.set(ruleIdx, toBeSummed + toSumTo);
+            }
+        }
+
+        public void addPercentages(int currentIndex) {
```

Comment:
```suggestion
        public void collectRulePercentage(int currentIndex) {
```
(adding might seem a sum)

Reply:
fixed